### DISCUSSION TITLE
Some Beginner Question - Session Cookies, GraphQL and Custom Servers

### DISCUSSION THREAD

**Discussion by codelover2k** [vercel/next.js#16381] [Help]
I am very new in Next.js comming from a React SPA and Apollo-Server background.
There are some questions that I have that would help me to jump into Next.js

1. How can I get session cookies in a Next.js server? (something like express-session)

2. How can I get apollo-server into Next.js and inject the context with the req.session.id ?

3. When do you need/recommend to add a custom server to Next.js?
    It looks like Next.js does not need that but I would like to know if there are some use cases and what they are.
    I also do not understand why you loose optimization if you use a custom server, because your custom server just gets the request first and then pass it to your Next.js portion.    

I hope you can help me with these questions, thanks!

**Comment by YoannBuzenet** [comment]
I have to say I don't see any case where you need a custom server.
